components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java

```diff
@@ -17,54 +17,24 @@
  */
 package org.wso2.carbon.identity.oauth.endpoint.user.impl;
 
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
 import org.apache.oltu.oauth2.common.utils.JSONUtils;
-import org.wso2.carbon.identity.application.common.model.ClaimMapping;
-import org.wso2.carbon.identity.oauth.cache.AuthorizationGrantCache;
-import org.wso2.carbon.identity.oauth.cache.AuthorizationGrantCacheEntry;
-import org.wso2.carbon.identity.oauth.cache.AuthorizationGrantCacheKey;
 import org.wso2.carbon.identity.oauth.endpoint.util.ClaimUtil;
-import org.wso2.carbon.identity.oauth.user.UserInfoClaimRetriever;
 import org.wso2.carbon.identity.oauth.user.UserInfoEndpointException;
 import org.wso2.carbon.identity.oauth2.dto.OAuth2TokenValidationResponseDTO;
 import org.wso2.carbon.identity.openidconnect.AbstractUserInfoResponseBuilder;
 
-import java.util.HashMap;
 import java.util.Map;
 
-import static org.apache.commons.collections.MapUtils.isEmpty;
-
 /**
  * Builds user info response as a JSON string according to
  * http://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse
  */
 public class UserInfoJSONResponseBuilder extends AbstractUserInfoResponseBuilder {
 
-    private static final Log log = LogFactory.getLog(UserInfoJSONResponseBuilder.class);
-
     @Override
     protected Map<String, Object> retrieveUserClaims(OAuth2TokenValidationResponseDTO tokenValidationResponse)
             throws UserInfoEndpointException {
-
-        Map<ClaimMapping, String> userAttributes = getUserAttributesFromCache(tokenValidationResponse);
-        Map<String, Object> claims;
-
-        if (isEmpty(userAttributes)) {
-            claims = ClaimUtil.getClaimsFromUserStore(tokenValidationResponse);
-            if (log.isDebugEnabled()) {
-                log.debug("User attributes not found in cache. Trying to retrieve from user store.");
-            }
-        } else {
-            UserInfoClaimRetriever retriever = UserInfoEndpointConfig.getInstance().getUserInfoClaimRetriever();
-            claims = retriever.getClaimsMap(userAttributes);
-        }
-
-        if (claims == null) {
-            claims = new HashMap<>();
-        }
-
-        return claims;
+        return ClaimUtil.getUserClaimsUsingTokenResponse(tokenValidationResponse);
```

Comment:
Can we not use the Utils. we can add this method into the base class of the Response Builder.
Utils takes out part of the knowledge to humans head.